pandas/tseries/tests/test_timeseries.py

```diff
@@ -2283,6 +2135,277 @@ def _simple_ts(start, end, freq='D'):
+        self.assertTrue(result.equals(expected))
+
+        today = datetime.today()
+        idx = Index([today], dtype=object)
+        result = idx.to_datetime()
+        expected = DatetimeIndex([today])
+        self.assertTrue(result.equals(expected))
+
+    def test_dataframe(self):
+
+        df = DataFrame({'year': [2015, 2016],
+                        'month': [2, 3],
+                        'day': [4, 5],
+                        'hour': [6, 7],
+                        'minute': [58, 59],
+                        'second': [10, 11],
+                        'ms': [1, 1],
+                        'us': [2, 2],
+                        'ns': [3, 3]})
+
+        result = to_datetime({'year': df['year'],
+                              'month': df['month'],
+                              'day': df['day']})
+        expected = Series([Timestamp('20150204 00:00:00'),
+                           Timestamp('20160305 00:0:00')])
+        assert_series_equal(result, expected)
+
+        # dict-like
+        result = to_datetime(df[['year', 'month', 'day']].to_dict())
+        assert_series_equal(result, expected)
+
+        # dict but with constructable
+        df2 = df[['year', 'month', 'day']].to_dict()
+        df2['month'] = 2
+        result = to_datetime(df2)
+        expected2 = Series([Timestamp('20150204 00:00:00'),
+                            Timestamp('20160205 00:0:00')])
+        assert_series_equal(result, expected2)
+
+        # unit mappings
+        units = [{'year': 'year',
+                  'month': 'month',
+                  'day': 'day',
+                  'hour': 'HH',
+                  'minute': 'MM',
+                  'second': 'SS'},
+                 {'year': '%Y',
+                  'month': '%m',
+                  'day': '%d',
+                  'hour': '%H',
+                  'minute': '%M',
+                  'second': '%S'},
+                 {'year': 'y',
+                  'month': 'month',
+                  'day': 'd',
+                  'hour': 'h',
+                  'minute': 'm',
+                  'second': 's'},
+                 ]
+
+        for d in units:
+            result = to_datetime(df[list(d.keys())].rename(columns=d))
+            expected = Series([Timestamp('20150204 06:58:10'),
+                               Timestamp('20160305 07:59:11')])
+            assert_series_equal(result, expected)
+
+        d = {'year': 'y',
+             'month': 'month',
+             'day': 'd',
+             'hour': 'h',
+             'minute': 'm',
+             'second': 's',
+             'ms': 'ms',
+             'us': 'us',
+             'ns': 'ns'}
+
+        result = to_datetime(df.rename(columns=d))
+        expected = Series([Timestamp('20150204 06:58:10.001002003'),
+                           Timestamp('20160305 07:59:11.001002003')])
+        assert_series_equal(result, expected)
+
+        # coerce back to int
+        result = to_datetime(df.astype(str), unit=d)
```

Comment:
I agree with Joris that we should restrict this method to very specific,
verbose names for each component.

On Tue, Apr 26, 2016 at 8:38 AM, Joris Van den Bossche <
notifications@github.com> wrote:

> In pandas/tseries/tests/test_timeseries.py
> <https://github.com/pydata/pandas/pull/12967#discussion_r61109133>:
>
> > +             'month': 'month',
> > +             'day': 'd',
> > +             'hour': 'h',
> > +             'minute': 'm',
> > +             'second': 's',
> > +             'ms': 'ms',
> > +             'us': 'us',
> > +             'ns': 'ns'}
> > +
> > +        result = to_datetime(df.rename(columns=d))
> > +        expected = Series([Timestamp('20150204 06:58:10.001002003'),
> > +                           Timestamp('20160305 07:59:11.001002003')])
> > +        assert_series_equal(result, expected)
> > +
> > +        # coerce back to int
> > +        result = to_datetime(df.astype(str), unit=d)
>
> yep, prob from the initial implementation which used unit
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/12967/files/e18c9cc7dcbc6e17c843517a05f348e878a68bcc#r61109133>
>